lib/srv/uacc/uacc.h

```diff
@@ -63,33 +79,37 @@ static int uacc_add_utmp_entry(const char *utmp_path, const char *wtmp_path, con
     entry.ut_pid = getpid();
     strncpy((char*) &entry.ut_host, hostname, sizeof(entry.ut_host));
     strncpy((char*) &entry.ut_user, username, sizeof(entry.ut_user));
-    entry.ut_session = 1;
+    entry.ut_session = getsid(0);
     entry.ut_tv.tv_sec = tv_sec;
     entry.ut_tv.tv_usec = tv_usec;
     memcpy(&entry.ut_addr_v6, &remote_addr_v6, sizeof(int32_t) * 4);
     errno = 0;
     setutent();
-    if (errno != 0) {
+    if (errno > 0) {
```

Comment:
It might be potentially useful to log the actual error - this should be extracted on the caller's side.